File: youlexuan-manager-web/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){

    //保存  ：添加和修改
    this.save = function(methodName,entity){
        return $http.post("../brand/"+methodName+".do",entity);
    }
    //查询所有
    this.findAll = function(){
        return $http.get("../brand/findAll.do");
    }
    //查询一个
    //$http.get('../brand/findOne.do?id='+id)
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }
    //删除
    //$http.get('../brand/delete.do?ids='+$scope.selectIds)
    this.dele = function(ids){
        return $http.get("../brand/delete.do?ids="+ids);
    }
    //搜索
    this.search = function(page,rows,searchEntity){
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,searchEntity);
    }

    /*this.findPage = function(page,rows){
        return $http.get("../brand/findPage.do?page="+page+"&rows="+rows);
    }

    this.update=function(entity){
        return $http.post("../brand/update.do",entity);
    }

    this.findById=function(id){
        return $http.get("../brand/findById.do?id="+id);
    }*/

    this.selectOptionList = function(){
        return $http.get("../brand/selectOptionList.do");
    }
});

File: youlexuansellergoodsinterface/src/main/java/com/youlexuan/sellergoods/service/BrandService.java
package com.youlexuan.sellergoods.service;

import com.youlexuan.entity.PageResult;
import com.youlexuan.entity.Result;
import com.youlexuan.pojo.good.Brand;

import java.util.List;
import java.util.Map;

/**
 *  品牌服务层接口
 */
public interface BrandService {
    /**
     * 返回全部列表
     * @return
     */
    public List<Brand> findAll();
    //返回分页列表
    public PageResult findPage(Integer page, Integer rows);
    //添加
    public Result insert(Brand brand);

    //返回一个品牌实体
    public Brand findOne(long id);
    //修改
    public void update(Brand brand);

    //批量删除
    public void delete(Long [] ids);
    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Brand brand, int pageNum,int pageSize);

    /**
     * 品牌下拉框数据
     */
    public List<Map> selectOptionList();


}
